### PR TITLE
Scaffolding for services

### PR DESCRIPTION
**Pull request by ASypula**

### CODE REVIEW

File: python-cache-demo/geo-service/geolocation_db_conn.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import mysql.connector
```
Comment (MA3CIN) [inline python-cache-demo/geo-service/geolocation_db_conn.py:1]: Nice. btw the dependency install is already taken care of for each service in the corresponding requirements.txt file (https://github.com/MA3CIN/resilient-six/blob/main/python-cache-demo/infra-service/requirements.txt), i tested it for infra-service.

File: python-cache-demo/geo-service/geolocation_db_conn.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import logging
+
+class GeolocationDBConnector:
+    def __init__(self, host, user, pwd, database="geolocation", logger=logging.Logger(__name__)):
+        self.logger = logger
+        self.logger.info(f"Connecting to the database {database} on the host: {host}.")
+        try:
+            self.db_conn = mysql.connector.connect(
+                host=host,
```
Comment (MA3CIN) [inline python-cache-demo/geo-service/geolocation_db_conn.py:10]: These are passed as ENV's (should be passed to geo_service.py when instantiating the DB Connector), also an example is shown in the infra-service

File: python-cache-demo/geo-service/geolocation_service.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    Get all registered devices.
+    """
+    logger.info(f"Getting information about device {device_id}.")
+    device = db.get_device(device_id)
+    response_dict = {"pos_x": device[3], "pos_y": device[4]}
+    return response_dict
+
+if __name__ == '__main__':
+    app.run(port=5000)
```
Comment (MA3CIN) [inline python-cache-demo/geo-service/geolocation_service.py]: Flask run options were changed for each service (host was defined, ports were changed) - please keep that in mind when merging!

File: python-cache-demo/infra-service/infra_service.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return db.get_users_devices(owner_id)
+
+@app.route('/devices', methods=['POST'])
+def register_device():
+    """
+    Registering a device.
+    Required input:
+      owner (int): id of the owner
+      model (int): id of the model
+      TODO: maybe we could provide an option to add a name of the model?
```
Comment (MA3CIN) [inline python-cache-demo/infra-service/infra_service.py]: We could. We need to agree on a data model ASAP, should be our first priority probably

File: python-cache-demo/stats-service/stats_db_conn.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import mysql.connector
+import logging
+
+class StatisticsDBConnector:
+    def __init__(self, host, user, pwd, database="statistics", logger=logging.Logger(__name__)):
+        self.logger = logger
+        self.logger.info(f"Connecting to the database {database} on the host: {host}.")
+        try:
+            self.db_conn = mysql.connector.connect(
+                host=host,
```
Comment (MA3CIN) [inline python-cache-demo/stats-service/stats_db_conn.py:10]: same as above, passed as ENVs

File: python-cache-demo/infra-service/infra_service.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    except Exception as e:
+        logger.error("Incorrect data provided for registering a device. Error: {e}.")
+        raise
+    try:
+      db.add_device(owner, model, x_pos, y_pos)
+      logger.info(f"Devide registered successfully.")
+    except Exception as e:
+      logger.error(f"Error when registering device. Error: {e}.")
+      raise
+    return jsonify(success=True)
```
Comment (MA3CIN) [inline python-cache-demo/infra-service/infra_service.py:130]: Smart, nice!

File: python-cache-demo/stats-service/stats_db_conn.py (hunk 5 above)
Comment (MA3CIN) [reply python-cache-demo/stats-service/stats_db_conn.py:10]: Ok, i see, the ENVS will be passed in stats_service.py

**Review by MA3CIN — COMMENTED** [review]
Really nice. I am sorry for pushing changes at 1 AM that will cause conflicts with this PR, i'll branch out from now on. Some minor comments from my side included, but overall really nice!

File: python-cache-demo/geo-service/geolocation_service.py (hunk 3 above)
Comment (ASypula) [reply python-cache-demo/geo-service/geolocation_service.py]: Updated ;)

File: python-cache-demo/stats-service/stats_db_conn.py (hunk 5 above)
Comment (ASypula) [reply python-cache-demo/stats-service/stats_db_conn.py:10]: Yes, they are passed when instantiating the class

File: python-cache-demo/geo-service/geolocation_db_conn.py (hunk 2 above)
Comment (ASypula) [reply python-cache-demo/geo-service/geolocation_db_conn.py:10]: Those params are passed in each <name>_service.py separately when establishing the connection.

**Comment by MA3CIN** [comment]
Looks good to me! Did you test it with the docker compose (service <-> service) and (service (infra) <-> database)? If so, then it's a LGTM from my side and we can merge

File: python-cache-demo/geo-service/geolocation_service.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    Get all registered devices.
+    """
+    logger.info("Getting all registered devices.")
+    return db.get_all_registered_devices()
+
+@app.route('/devices/position/<int:device_id>', methods=['GET'])
+def get_device_position(device_id):
+    """
+    Get all registered devices.
```
Comment (WasilewM) [inline python-cache-demo/geo-service/geolocation_service.py]: Update the function description please

File: python-cache-demo/geo-service/geolocation_service.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+# # get endpoint IF cache is empty. 
+# #invalidate cache if you have a serial number without a corresponding cached owner 
+# #with cache enabled
+# def get_owners():
+#    if (INFRA_URL in session.cache.urls()):
+#       print("gotten from cache")
+#    else:
+#       print("gotten manually")
+#    return session.get(INFRA_URL)
```
Comment (WasilewM) [inline python-cache-demo/geo-service/geolocation_service.py]: Do we want to leave the commented code?

File: python-cache-demo/infra-service/infra_service.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Get all registered devices.
+    """
+    logger.info("Getting all registered devices.")
+    return db.get_all_registered_devices()
+
+@app.route('/devices/position/<int:device_id>', methods=['GET'])
+def get_device_position(device_id):
+    """
+    Get all registered devices.
```
Comment (WasilewM) [inline python-cache-demo/infra-service/infra_service.py]: Update the function description please

File: python-cache-demo/geo-service/geolocation_service.py (hunk 7 above)
Comment (ASypula) [reply python-cache-demo/geo-service/geolocation_service.py]: Updated :)

File: python-cache-demo/endpoints.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
+## Available endpoints per service
```
Comment (WasilewM) [inline python-cache-demo/endpoints.md:1]: General comments to this file - looks good to me

**Comment by MA3CIN** [comment]
So far it's looking great! I am very happy to see a Logger - finally we can  collect useful logs :)

**Pull request merged**